nowriter/tmencoding/tm_encoder_builder.go

```diff
@@ -0,0 +1,32 @@
+package tmencoding
+
+import "time"
+
+// Allow chaining builder pattern syntactic sugar variation of Facade
+// teb.WriteInt8(42).WriteInt32(my_integer).Bytes()
+type TMEncoderBuilderIntr interface {
+	Bytes() []byte
+	WriteBool(b bool) TMEncoderBuilderIntr
+	WriteFloat32(f float32) TMEncoderBuilderIntr
+	WriteFloat64(f float64) TMEncoderBuilderIntr
+	WriteInt8(i int8) TMEncoderBuilderIntr
+	WriteInt16(i int16) TMEncoderBuilderIntr
+	WriteInt32(i int32) TMEncoderBuilderIntr
+	WriteInt64(i int64) TMEncoderBuilderIntr
+	WriteOctet(b byte) TMEncoderBuilderIntr
+	WriteOctets(b []byte) TMEncoderBuilderIntr
+	WriteTime(t time.Time) TMEncoderBuilderIntr
+	WriteUint8(i uint8) TMEncoderBuilderIntr
+	WriteUint16s(iz []uint16) TMEncoderBuilderIntr
+	WriteUint32(i uint32) TMEncoderBuilderIntr
+	WriteUint64(i uint64) TMEncoderBuilderIntr
+	WriteUvarint(i uint) TMEncoderBuilderIntr
+	WriteVarint(i int) TMEncoderBuilderIntr
+}
+
+// Ensure chaining syntax functions properly with compile-time assertion.
```

Comment:
so the compiler doesn't complain about `confirm_syntax` being unused. and the reason i don't use `_` is because i want to give the reader a hint why i'm doing it. maybe i should add another hint.